enhancements/installer/assisted-installer-bare-metal-validations.md

```diff
@@ -0,0 +1,280 @@
+in user-provisioned workflows.
+
+#### Story 2
+
+As a developer of the OpenShift installer, I want to be able to add
+validation rules for bare metal clusters correctly based on whether
+the cluster is being built using the installer-provisioned
+infrastructure workflow or the user-provisioned workflow so I can
+apply only the rules necessary for each case.
+
+### Implementation Details/Notes/Constraints [optional]
+
+All of the validation is currently performed during early phases of
+the installer for the `InstallConfig` target. The user-provisioned
+workflow used by the assisted installer does not run the phase for the
+`Cluster` target (it only ever runs up to the `IgnitionConfigs`
+target), so if we move some of the validation into the later phase we
+will have a better separation of the parts needed to build the assets
+for the user-provisioned workflow and the assisted installer from the
+parts needed to run the installer for installer-provisioned
+infrastructure for bare metal. A secondary benefit is that the
+validation for the `baremetal` platform will more closely match the
+validation for cloud platforms.
+
+The install config inputs are currently validated as part of
+generating the `InstallConfig` asset in
+[InstallConfig.finish()](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/installconfig.go#L141),
+which invokes some logic that applies to all platforms in
+[ValidateInstallConfig()](https://github.com/openshift/installer/blob/master/pkg/types/validation/installconfig.go#L40)
+before running the platform-specific logic from
+[validatePlatform()](https://github.com/openshift/installer/blob/master/pkg/types/validation/installconfig.go#L341). For
+the `baremetal` platform that results in calling
+[ValidatePlatform()](https://github.com/openshift/installer/blob/master/pkg/types/baremetal/validation/platform.go#L218).
+
+The
+[PlatformProvisionCheck](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/platformprovisioncheck.go)
+and
+[PlatformCredsCheck](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/platformcredscheck.go)
+assets are both dependencies for the `Cluster` asset, which means they
+are evaluated later in the installation process. Both already have
+platform-specific logic, although not for the `baremetal` platform.
+
+The assisted installer does not need the `bootstrapProvisioningIP`,
+`provisioningNetworkCIDR`, `provisioningDHCPRange`, or
+`provisioningNetworkInterface` fields. The values in those fields are
+used to configure the provisioning tools used for the
+installer-provisioned infrastructure workflow, so those validation
+checks could move to a new function in the
+`pkg/types/baremetal/validation` package that can be invoked from
+[PlatformProvisionCheck.Generate()](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/platformprovisioncheck.go#L37).
+
+There are two validation rules for the `clusterOSImage` and
```

Comment:
what are those used for? just for general knowledge

Reply:
They are used to optionally specify a URL for the two RHCOS images needed for an IPI baremetal deployment - these are large and by default are downloaded every deploy from the internet, so this interface enables faster repeated testing and also deployment in disconnected environments.